### ISSUE TITLE
Absl subproject dos not respect libdir

### ISSUE THREAD

**Issue by DanySpin97** [joshkunz/ashuffle#57]
meson arguments:

```meson --prefix=/usr --bindir=/usr/x86_64-pc-linux-musl/bin --sbindir=/usr/x86_64-pc-linux-musl/bin --libdir=/usr/x86_64-pc-linux-musl/lib --libexecdir=/usr/x86_64-pc-linux-musl/libexec --includedir=/usr/x86_64-pc-linux-musl/include --sysconfdir=/etc --datadir=/usr/share --mandir=/usr/share/man --default-library=shared --native-file meson-native.txt --wrap-mode=nodownload /tmp/paludis/build/media-sound-ashuffle-3.3.0/work/ashuffle-3.3.0```

From meson logs:
```
Installing subprojects/absl/librandom_internal_distribution_test_util.a to /tmp/paludis/build/media-sound-ashuffle-3.3.0/image//usr/lib64
```

**Comment by joshkunz** [comment]
Hm... That's not good. absl really shouldn't be getting installed at all. absl doesn't really support dynamic linking, and if absl is statically linked it doesn't need to be installed. In the log-line you pasted a static absl library is being installed, which is definitely wrong. So something is definitely broken in our configuration.

I'll take a look at this.

**Comment by brainpower** [comment]
I've investigated this, too. I came up with this meson issue, which seems related:
https://github.com/mesonbuild/meson/issues/2550

There's probably nothing that could be done from ashuffle's side...

**Comment by joshkunz** [comment]
Thanks for the link @brainpower, but I think there's another issue going on here.

In the generated `meson.build` built by the cmake module, all the absl libraries are set with `install: true`. Looking at the meson source, the cmake module pulls this info [straight from cmake](https://github.com/mesonbuild/meson/blob/master/mesonbuild/cmake/fileapi.py#L168), based on whether or not there is an "install" rule for that library.

The issue is that absl is generating these install rules when we don't want it to. absl's `CMakeLists.txt` decides to generate install rules depending on whether or not it is being included  using `add_subdirectory` or built by itself. It detects this by [comparing the source root with the "project" root](https://github.com/abseil/abseil-cpp/blob/master/CMakeLists.txt#L43-L47), which would presumably not match if absl was included in another project. Since meson runs cmake on absl in isolation, absl thinks it's being built stand-alone and generates these install rules.

I've mailed out a PR to get around this (in #59). It works by tweaking the project source root to match ashuffle's root rather than absls, preventing the creation of the install rules. It prevents the install when I test locally.

**Comment by joshkunz** [comment]
This should be fixed as of v3.4.0.
